src/functor.jl

```diff
@@ -29,6 +29,35 @@ macro functor(args...)
   functorm(args...)
 end
 
+function makeflexiblefunctor(m::Module, T, pfield)
+  pfield = QuoteNode(pfield)
+  @eval m begin
+    $Functors.functor(::Type{<:$T}, x) = begin
+      function re(y)
+        all_args = map(fieldnames($T)) do fn
+          field = fn in getfield(x, $pfield) ? getfield(y, fn) : getfield(x, fn)
+          return field
+        end
+        return $T(all_args...)
+      end
+      func = NamedTuple{pfields}(map(p -> getfield(x, p), pfields))
```

Comment:
```suggestion
      func = NamedTuple{pfield}(map(p -> getfield(x, p), pfield))
```

Reply:
`pfield` is a symbol. We want the value that symbol indexes to, not the symbol itself.

Reply:
I think ci didn't like it, but of course, we need the value here, so that field needs to be closed over. I'll let @SBuercklin comment on it.

Reply:
The reason CI doesn't like it is because it requires the other suggestion https://github.com/FluxML/Functors.jl/pull/19#discussion_r675715573. I didn't want to mark the entire function as one because GH isn't smart enough to handle the diff, but they're technically intertwined.